github_pr.py

```diff
@@ -108,12 +109,64 @@ def github_list_prs(**args):
     elif 'head' in args and args['head']:
         prs = _load_prs_by_branch(**args)
         _print_prs(prs, **args)
+    elif 'filter' in args and args['filter']:
+        prs = github_filter_prs(**args)
+        #_print_prs(prs, **args)
     else:
         prs = repo.get_pulls()
         _print_prs(prs, **args)
     return list_return_obj
 
 
+def github_filter_prs(**args):
+    """
+    Filters prs to return only what is contained in the filters
+    """
+    gh = Github(args['token'])
+    repo = gh.get_repo(args['repo'])
+    all_prs = [{'pr':pr} for pr in repo.get_pulls()]
```

Comment:
all_prs is a list of dictionaries. since we need access to different information in different object types (pr vs issue) we decided to connect them with a dictionary. So at first we do a get of the pr objects, since they are smaller compared to the issue objects and then we only get the issue object for that PR number if we actually need information from outside the PR object.